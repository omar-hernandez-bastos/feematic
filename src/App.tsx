import Calculator from 'pages/Calculator'
import { useEffect, useState } from 'react'
import { getRates } from 'services/axios'

function App() {
  const [cop, setCop] = useState(3750)

  const [ves, setVes] = useState(4.63)

  const fetchRates = async () => {
    const { cop, ves } = await getRates()
    setCop(+cop)
    setVes(+ves)
  }
  useEffect(() => {
    fetchRates()
  }, [])

  return (
    <div className="flex justify-center items-center min-h-screen bg-[#DAF0FF] min-w-screen">
      <Calculator cop={cop} ves={ves} />
    </div>
  )
}

export default App
