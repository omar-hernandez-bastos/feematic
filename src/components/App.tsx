import Calculator from 'pages/Calculator'
import { useEffect, useState } from 'react'

function App() {
  const [USD_TO_COP, SET_USD_TO_COP] = useState(3750)

  const [USD_TO_VEF, SET_USD_TO_VEF] = useState(4.63)

  const COP_TO_USD = 1 / USD_TO_COP
  const COP_TO_VEF = COP_TO_USD * USD_TO_VEF
  const VEF_TO_USD = 1 / USD_TO_VEF
  const VEF_TO_COP = VEF_TO_USD * USD_TO_COP
  useEffect(() => {
    const requestOptions = {
      method: 'GET',
      redirect: 'follow' as const
    }

    fetch(
      'https://openexchangerates.org/api/latest.json?app_id=5fe7bbbc6f2f4500a692a133235cdd88',
      requestOptions
    )
      .then((response) => response.json())
      .then((result) => {
        const { COP, VES } = result.rates
        SET_USD_TO_COP(COP)
        SET_USD_TO_VEF(VES)
      })
      .catch((error) => console.log('error', error))
  }, [])
  const propsCalculator = {
    USD_TO_COP,
    USD_TO_VEF,
    VEF_TO_COP,
    VEF_TO_USD,
    COP_TO_VEF,
    COP_TO_USD
  }
  return (
    // eslint-disable-next-line tailwindcss/no-custom-classname
    <div className="flex justify-center items-center min-h-screen bg-[#DAF0FF] min-w-screen">
      <Calculator {...propsCalculator} />
    </div>
  )
}

export default App
