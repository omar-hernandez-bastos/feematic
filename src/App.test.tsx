import { render } from '@testing-library/react'
import App from './App'

describe('App Test', () => {
  it('App Component should render correctly', () => {
    const AppProps = {}
    const { container } = render(<App {...AppProps} />)
    expect(container).toMatchSnapshot()
  })
})
