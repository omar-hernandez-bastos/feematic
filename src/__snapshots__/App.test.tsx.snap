// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test App Component should render correctly 1`] = `
<div>
  <div
    class="flex justify-center items-center min-h-screen bg-[#DAF0FF] min-w-screen"
  >
    <div
      class="overflow-y-auto w-screen md:w-96 h-screen md:h-[90vh] bg-[#303136] md:rounded-3xl md:shadow-2xl md:shadow-cyan-500/50"
    >
      <div
        class="flex flex-col justify-between p-8 w-full h-auto"
      >
        <div
          class="flex flex-col justify-center items-end w-full h-[40vh]"
        >
          <div
            class="text-5xl font-bold text-right text-white"
          >
            0,00
            ðŸ‡ºðŸ‡¸
          </div>
          <div
            class="flex flex-col my-2 text-3xl font-light text-right text-gray-300"
          >
            <span>
              0,00 ðŸ‡¨ðŸ‡´
            </span>
            <span>
              0,00 ðŸ‡»ðŸ‡ª
            </span>
          </div>
        </div>
        <div
          class="grid grid-cols-4 gap-3 pb-12 w-full h-[50vh]"
        >
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              color="cyan-200"
              data-testid="ðŸ‡¨ðŸ‡´"
            >
              ðŸ‡¨ðŸ‡´
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center shadow-2xl shadow-cyan-500/50"
              color="gray-100"
              data-testid="ðŸ‡ºðŸ‡¸"
            >
              ðŸ‡ºðŸ‡¸
            </button>
          </div>
          <div
            class="row-span-2"
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="ðŸ‡»ðŸ‡ª"
            >
              ðŸ‡»ðŸ‡ª
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="7"
            >
              7
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="8"
            >
              8
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="9"
            >
              9
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="4"
            >
              4
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="5"
            >
              5
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="6"
            >
              6
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="1"
            >
              1
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="2"
            >
              2
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="3"
            >
              3
            </button>
          </div>
          <div
            class="row-span-2"
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="ðŸš«"
            >
              ðŸš«
            </button>
          </div>
          <div
            class="col-span-2"
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="0"
            >
              0
            </button>
          </div>
          <div
            class=""
          >
            <button
              class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
              data-testid="."
            >
              .
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
